File: C2_gestion_user/objects/bateau/carnet_maintenance/Carnet_maintenance.java
package C2_gestion_user.objects.bateau.carnet_maintenance;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Class d'objet Carnet_maintenance
 * - contient un objet entretien
 * - contient un boolean déterminant si l'entretien a été fait
 */

public class Carnet_maintenance {

    public static void main(String[] args) {
    }

    private boolean entretienfait;
    private Entretien entretien;

    public Carnet_maintenance(boolean entretienfait, String modele) {
        this.entretien = new Entretien(modele);
        this.entretienfait = entretienfait;
    }

    public boolean isEntretienfait() {
        return entretienfait;
    }
    public void setEntretienfait(boolean entretienfait) {
        this.entretienfait = entretienfait;
    }
    public Entretien getEntretien() {
        return entretien;
    }
    public void setEntretien(Entretien entretien) {
        this.entretien = entretien;
    }

    @Override
    public String toString() {
        return "Carnet_maintenance{" +
                "entretienfait=" + entretienfait +
                ", entretien=" + entretien +
                '}';
    }
    public JSONObject toJson() {
        JSONObject jo = new JSONObject();
        try {
            jo.put("entretienfait", this.entretienfait);
            jo.put("entretien", this.entretien.toJson());
        }
        catch (JSONException e) {
            System.out.println("erreur "+e);
        }
        return jo;
    }
}

File: C1_portail/objects/carnet_maintenance/Carnet_maintenance.php
<?php
/**
 * Class d'objet Carnet_maintenance
 * - contient un objet entretien
 * - contient un boolean déterminant si l'entretien a été fait
 * http://localhost/C1_portail/objects/carnet_maintenance/Carnet_maintenance.php
 */

require_once('Categorie_Entretien.php');
require_once('Entretien.php');

class Carnet_maintenance implements \JsonSerializable
{
    private $entretienfait;
    private $entretien;

    /**
     * Carnet_maintenance constructor.
     * @param $entretienfait
     * @param $modele
     */
    public function __construct($entretienfait, $modele) {
        $this->entretien = new Entretien($modele);
        $this->entretienfait = $entretienfait;
    }

    /**
     * @return mixed
     */
    public function getEntretienfait()
    {
        return $this->entretienfait;
    }
    /**
     * @param boolean $entretienfait
     */
    public function setEntretienfait($entretienfait): void
    {
        $this->entretienfait = $entretienfait;
    }

    /**
     * @return Entretien
     */
    public function getEntretien(): Entretien
    {
        return $this->entretien;
    }

    /**
     * @param Entretien $entretien
     */
    public function setEntretien(Entretien $entretien): void
    {
        $this->entretien = $entretien;
    }


    /**
     * @return string
     */
    public function __toString() {
        return  '*Carnet Maintenance object<br/>'.
                'entretienfait= '.$this->entretienfait.'<br/>'.
                'entretien= <br/>'.$this->entretien;
    }

    public function jsonSerialize() {
        return [
            'entretienfait' => $this->getEntretienfait(),
            'entretien' => $this->getEntretien()->jsonSerialize()
        ];
    }
}